### ISSUE TITLE
NetworkAclRule sets attributes when unneeded

### ISSUE THREAD

**Issue by rirze** [hashicorp/terraform-cdk#1695]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

cdktf 0.10.1
cdktf_cdktf_aws_provider @ latest Python package

### Affected Resource(s)

`vpc.NetworkAclRule` in regards to `to_port` and `from_port`. 

### Expected Behavior

Terraform output should not contain `to_port` and `from_port` when it is unspecified for `protocol='icmp'`.

### Actual Behavior

Terraform output sets dummy values for these parameters which then causes a recreate for existing resources.  

### Steps to Reproduce
```python
nacl_rule = NetworkAclRule(
       	some_scope, some_id, 
		network_acl_rule='nacl-00000000',
        rule_number=100,
		egress=False,
		protocol='icmp',
		cidr_block='10.0.0.8/32',
		icmp_code=-1,
		icmp_type=-1,
		rule_action='allow',
)
```                                          

### Important Factoids

From my understanding, `to_port` and `from_port` are only required when protocol is `tcp` or `udp`. (ref https://docs.aws.amazon.com/cli/latest/reference/ec2/create-network-acl-entry.html#options). However, when not specifying `to_port` and `from_port` in the `cdktf` `NetworkAclRule` object, it sets `to_port-1` and `from_port=-1` in the Terraform output:

```json
,
      "some_id": {
        "cidr_block": "10.0.0.8/32",
        "egress": false,
        "from_port": -1,
        "icmp_code": -1,
        "icmp_type": -1,
        "network_acl_id": "${aws_network_acl.some_nacl.id}",
        "protocol": "icmp",
        "rule_action": "allow",
        "rule_number": 100,
        "to_port": -1
      },
```

Which ends up causing a recreate when importing existing NACL rules that allow ICMP to all ports. If it didn't have the `to_port` and `from_port` attributes, I believe it wouldn't recreate the resource.

### Notes
I'm unsure if this is an issue with `cdktf` or Terraform itself. Maybe an issue with the aws provider schema itself?

**Comment by rirze** [comment]
This could be temporarily fixed by implementing #1696

**Comment by rirze** [comment]
🤦 
I was looking at the wrong resource. It is correctly omitting the keys in the proper Terraform resources.
